src/python/pants/rules/core/filedeps.py

```diff
@@ -16,28 +17,47 @@ class Filedeps(LineOriented, Goal):
   closure of targets are also included.
   """
 
-  # TODO: Until this implements more of the options of `filedeps`, it can't claim the name!
-  name = 'fast-filedeps'
+  name = 'filedeps'
+
+  @classmethod
+  def register_options(cls, register):
+    super().register_options(register)
+    register(
+      '--absolute', type=bool, default=True,
+      help='If True, output with absolute path; else, output with path relative to the build root'
+    )
+    register(
+      '--globs', type=bool,
+      help='Instead of outputting filenames, output globs (ignoring excludes)'
+    )
 
 
 @console_rule
 def file_deps(
   console: Console,
   filedeps_options: Filedeps.Options,
+  build_root: BuildRoot,
```

Comment:
So... one development since this was opened is the `Workspace` type.

I _think_ that the semantics of the `BuildRoot` type (exposes some information) are sufficiently different from the `Workspace` type (implies mutable access) that having them separate could be reasonable. I could also see merging them, and exposing `Workspace.root` or something. "Relatively small single purpose things to DI" is a thing... but the `BuildRoot` is almost too specific a thing, and would always have a subset of the fields of `Workspace`.

Fine either way though.

Reply:
cc @gshuflin